bare_metal_tests/TESTS/bringup/uart/test.py

```diff
@@ -0,0 +1,330 @@
+  serial connection speed equal to 9600 b/s.
+  
+  For more options:
+      test.py -h
+      
+"""
+
+serial_port = None
+RETRY_LIMIT = 10
+
+# List of expected messages from the mbed device.
+message_list = ['x\0', 
+                'mbed\0',
+                'Green Tea UART bear metal test.\0', 
+                'Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message--.\0',
+                'PASSED\0']
+
+def exit():
+    if(serial_port != None and serial_port.isOpen() == True):
+        serial_port.close()
+    sys.exit()
+        
+def create_test_header_file(baudrate):
+    """! Create header file with the specified baud rate.
+    
+    @param baudrate specified transmission baud rate.
+    """
+    file = open('test_config.h','w') 
+    file.write('/* This file has been automatically generated by the python script. */\n')
+    file.write('/* This file contains serial port baud rate. */\n')
+    file.write('#define BAUD_RATE %s\n' % baudrate)
+    file.close()
+
+def run_command(cmd):
+    """! Runs command and prints proc stdout on screen.
+         It is used to build the test.
+         
+    @param cmd command to be executed.
+    @return image path.
+    """
+    image_path = str()
+    
+    # Remove build log file if exists
+    try:
+        os.remove('build.log')
+    except OSError:
+        pass
+    
+    try:
+        p = Popen(cmd, stdout=sys.stdout)
+    except OSError as e:
+        logging.error('Command can not be executed.')
+        exit()
+
+    # Check process status.
+    returncode = p.wait()
+    if returncode < 0:
+        logging.error('Error while executing command.')
+        exit()
+    
+    # Get image path     
+    try:
+        with open('build.log') as f:
+            print('file ok')
+            build_data = json.load(f)
+            print('json ok')
+            image_path = build_data['builds'][0]['bin']
+            print('json ok')
+    except:
+        logging.error('Unable to locate image file.')
+        exit()
+       
+    return image_path
+    
+def reset_mbed_board(port, delay_reset):
+    """Perform reset of the mbed device by sending
+       a break signal to the device.
+       
+    @param port serial port id.
+    @param delay_reset delay time after reset [s].
+    """
+    logging.info('Reset the mbed board...')
+ 
+    break_port = serial.Serial(port=port, baudrate=9600)
+    
+    break_port.sendBreak()
+
+    break_port.close()
+    
+    logging.info('Wait %f sec after the reset.' % (delay_reset))
+    time.sleep(delay_reset)
+    
+def flash_mbed_board(src, dst, delay):
+    """Perform flash of the mbed device.
+    
+    @param src path to the .bin file.
+    @param dst path to mbed device.
+    @param delay delay time after flashing [s].
+    """
+    logging.info('Flashing mbed board...')
+    logging.info('source: %s' % (src))
+    logging.info('destination: %s' % (dst))
+    
+    shutil.copyfile(src, dst)
```

Comment:
Can we reliably flash a new target by copying in this manner?

Reply:
The new version assumes that user performs board flashing manually. Flashing function does not exist in the new version.